torch/testing/_core.py

```diff
@@ -275,6 +276,111 @@ def make_non_contiguous(tensor: torch.Tensor) -> torch.Tensor:
+
+    If :attr:`low` and :attr:`high` are not passed, following default values are considered depending on the given
+    :attr:`dtype`:
+
+    - boolean type: `low` = 0, `high` = 2.
+    - uint8 type: `low` = 0, `high` = 9.
+    - floating and integral types: `low` = -9 and `high` = 9.
+    - complex types, for each real and imaginary part: `low` = -9, `high` = 9.
+
+    If :attr:`low` and :attr:`high` are passed, they are considered only if they are within the
+    limit of the :attr:`dtype`. Following are a few conditions that are taken care of:
+
+    - If :attr:`low` and/or :attr:`high` are specified and within dtype limits: the values are taken as they were.
+    - If :attr:`low` and/or :attr:`high` are specified but exceed the limits: :attr:`dtype` limits are considered instead.
+    - If :attr:`low` is ``-inf`` and/or :attr:`high` is ``inf``: :attr:`dtype` limits are considered instead
+    - If :attr:`low` is ``inf`` or ``nan`` and/or :attr:`high` is ``-inf`` or nan: a `ValueError` is raised, since these are invalid values for the range of output tensor.
+
+    If :attr:`noncontiguous` is ``True``, a noncontiguous tensor with the given size will be returned unless the
+    size specifies a tensor with a 1 or 0 elements in which case the noncontiguous parameter is ignored because
+    it is not possible to create a noncontiguous Tensor with a single element.
+
+    If :attr:`exclude_zero` is ``True`` (default is ``False``), all the values matching to zero in
+    the created tensor are replaced with a ``tiny`` (smallest positive representable number) value if floating type,
+    [``tiny`` + ``tiny``.j] if complex type and ``1`` if integer/boolean type.
+    """
+    def _modify_low_high(low, high, lowest, highest, default_low, default_high, dtype):
+        """
+        Modifies (and raises ValueError when appropriate) low and high values given by the user (input_low, input_high) if required.
+        """
+        def clamp(a, l, h):
+            return min(max(a, l), h)
+
+        low = low if low is not None else default_low
+        high = high if high is not None else default_high
+
+        # Checks for error cases
+        if low != low or high != high:
+            raise ValueError("make_tensor: one of low or high was NaN!")
+        if low > high:
+            raise ValueError("make_tensor: low must be weakly less than high!")
+
+        low = clamp(low, lowest, highest)
+        high = clamp(high, lowest, highest)
+
+        if dtype in integral_types():
+            return math.floor(low), math.ceil(high)
+
+        return low, high
+
+    if dtype is torch.bool:
+        result = torch.randint(0, 2, size, device=device, dtype=dtype)
+    elif dtype is torch.uint8:
+        ranges = (torch.iinfo(dtype).min, torch.iinfo(dtype).max)
+        low, high = _modify_low_high(low, high, ranges[0], ranges[1], 0, 9, dtype)
+        result = torch.randint(low, high, size, device=device, dtype=dtype)
+    elif dtype in integral_types():
```

Comment:
Since the dtype getters will be internalized (see #63554), I suggest we simply hardcode all the supported dtypes here. This should also be applied to the floating and complex conditions.

Reply:
Thanks for pointing this out, @pmeier - should be fixed in the recent commit.